src/features/unified-share-modal/UnifiedShareForm.js

```diff
@@ -540,28 +540,53 @@ class UnifiedShareForm extends React.Component<USFProps, State> {
     }
 
     renderUpgradeLinkDescription() {
-        const { openUpgradePlanModal = () => {}, trackingProps = {} } = this.props;
+        const { openUpgradePlanModal = () => {}, showNewUpgradeText = false, trackingProps = {} } = this.props;
         const { inviteCollabsEmailTracking = {} } = trackingProps;
         const { upgradeLinkProps = {} } = inviteCollabsEmailTracking;
 
+        let description = messages.upgradeGetMoreAccessControlsDescription;
+        let messageValuesObj = {
+            upgradeGetMoreAccessControlsLink: (
+                <PlainButton
+                    className="upgrade-link"
+                    onClick={openUpgradePlanModal}
+                    type="button"
+                    {...upgradeLinkProps}
+                >
+                    <FormattedMessage {...messages.upgradeGetMoreAccessControlsLink} />
+                </PlainButton>
+            ),
+        };
+
+        if (showNewUpgradeText) {
+            description = messages.upgradeCollaboratorAccessDescription;
+            messageValuesObj = {
+                upgradeGetMoreAccessControlsLink: (
+                    <PlainButton
+                        className="upgrade-link"
+                        data-resin-target="external_collab_newcopy_upgrade_cta"
+                        onClick={openUpgradePlanModal}
+                        type="button"
+                    >
+                        <FormattedMessage {...messages.upgradeLink} />
+                    </PlainButton>
+                ),
+                collaboratorAccessLink: (
+                    <Link
+                        className="upgrade-link"
+                        href="https://support.box.com/hc/en-us/articles/360044196413-Understanding-Collaborator-Permission-Levels"
+                        target="_blank"
+                    >
+                        <FormattedMessage {...messages.collaboratorAccessLink} />
+                    </Link>
+                ),
+            };
+        }
+
         return (
             <div className="upgrade-description">
                 <UpgradeBadge />
-                <FormattedMessage
-                    values={{
-                        upgradeGetMoreAccessControlsLink: (
-                            <PlainButton
-                                className="upgrade-link"
-                                onClick={openUpgradePlanModal}
-                                type="button"
-                                {...upgradeLinkProps}
-                            >
-                                <FormattedMessage {...messages.upgradeGetMoreAccessControlsLink} />
-                            </PlainButton>
-                        ),
-                    }}
-                    {...messages.upgradeGetMoreAccessControlsDescription}
-                />
+                <FormattedMessage values={messageValuesObj} {...description} />
```

Comment:
is this convention to grab the FormattedMessage props like this ? Or would it be better to save the FormattedMessage as a parameter once we've determined which message we want and use that parameter here instead?